### PR TITLE
Fixed isOfType helper

### PR DESCRIPTION
**Pull request by kotarella1110**
fixed https://github.com/thebrodmann/deox/pull/127#discussion_r354848344, https://github.com/thebrodmann/deox/pull/127#discussion_r355155500.

### CODE REVIEW

File: src/is-of-type.ts [hunk 1]
```diff
@@ -9,21 +9,17 @@ import { ExtractAction } from './types'
...
- *   '[Counter] reset',
- *   resolve => (value: number) => resolve(value)
- * )
- * isOfType(increment.type, increment()) //=> true
+ * isOfType('[Counter] increment', increment()) //=> true
  * @example
- * isOfType([increment.type, decrement.type], increment()) //=> true
+ * isOfType(['[Counter] increment', '[Counter] decrement'], increment()) //=> true
  * @example
  * isOfType(decrement(), increment()) //=> false
```
Comment (thebrodmann) [inline src/is-of-type.ts]: This example is not correct.

File: src/utils/data.ts [hunk 2]
```diff
@@ -1,6 +1,8 @@
 export const merge = <T extends {}>(...objs: T[]): T =>
   Object.assign({}, ...objs)
 
-export function castArray<TValue>(value: TValue | TValue[]): TValue[] {
+export function castArray<TValue>(
+  value: TValue | ReadonlyArray<TValue>
+): ReadonlyArray<TValue> {
```
Comment (thebrodmann) [inline src/utils/data.ts]: We are returning a mutable array. So let's return `TValue[]`.

Comment (kotarella1110) [reply src/utils/data.ts]: Sorry! I will fix this soon.

File: src/is-of-type.ts (hunk 1 above)
Comment (kotarella1110) [reply src/is-of-type.ts]: Why is this not correct?

Comment (thebrodmann) [reply src/is-of-type.ts]: My bad 🤦🏻‍♂️

File: src/utils/data.ts (hunk 2 above)
Comment (kotarella1110) [reply src/utils/data.ts]: completed! https://github.com/thebrodmann/deox/pull/132/commits/5210e82f593533bb9ba57ae85e668384e93b3b44

**Comment by thebrodmann** [comment]
I highly appreciate your support @kotarella1110.

**Pull request merged**